controllers/applicationvolumereplication_controller.go

```diff
@@ -295,39 +303,99 @@ func (r *ApplicationVolumeReplicationReconciler) extractHomeClusterAndPeerCluste
 	return homeCluster, peerCluster, nil
 }
 
-func (r *ApplicationVolumeReplicationReconciler) doCreateOrUpdateManifestWork(name string, namespace string,
-	homeCluster string) error {
-	r.Log.Info("attempt to create and update ManifestWork")
+func (r *ApplicationVolumeReplicationReconciler) createOrUpdateVRGRolesManifestWork(namespace string) error {
+	// TODO: Enhance to remember clusters where this has been checked to reduce repeated Gets of the object
```

Comment:
If we decide to merge this, need to track the above in an issue and also need to ensure this is the right manner of delivering the `ClusterRoles` and `ClusterRoleBindings`, rather than as a policy(?) to all `ManagedClusters`.

Reply:
I think we can merge it and create an issue to track it. I think it is fine and I only have one issue doing it this way, which is, how do we ensure that the *rolebindings is delivered before the VRG? I think we can always create a VRG only after we make sure that the state of the *RoleBindings MW is `Applied`.

Reply:
Klusterlet fails to create VRG and retries the reconciliation of manifestWork, and succeeds once the roles are updated. So order is not mandatory (yet?) in this case.

I did think of pushing Roles to all MCs as some form of init routine, but not all MCs would be in a DR relationship and hence do not need these roles. So there are other alternatives than just letting the system reach healthy state eventually.